src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,157 @@
+    private Optional<Double> lastControlledTime;
+
+    public enum IntakePosition {
+        DEPLOYED,
+        STOWED,
+        CLEAR_OF_CONVEYOR
+    }
+
+    private TreeMap<IntakePosition, Double> intakePositionMap;
+    private IntakePosition latestPosition = IntakePosition.STOWED;
+    private IntakePosition desiredPosition = IntakePosition.STOWED;
+
+    public Intake() {
+        initPivotMotor();
+        initIntakeTalons();
+        intakePositionMap = new TreeMap<IntakePosition, Double>();
+        intakePositionMap.put(IntakePosition.DEPLOYED, IntakeCal.INTAKE_DEPLOYED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.STOWED, IntakeCal.INTAKE_STOWED_POSITION_DEGREES);
+        intakePositionMap.put(IntakePosition.CLEAR_OF_CONVEYOR, IntakeCal.INTAKE_SAFE_POSITION_DEGREES);
+        lastControlledTime = Optional.of(Timer.getFPGATimestamp());
+    }
+
+    public void initPivotMotor() {
+        pivotController = new ProfiledPIDController(IntakeCal.INTAKE_PIVOT_P,
+            IntakeCal.INTAKE_PIVOT_I, IntakeCal.INTAKE_PIVOT_D, new TrapezoidProfile.Constraints(
+                    IntakeCal.PIVOT_MAX_VELOCITY_DEG_PER_SECOND,
+                    IntakeCal.PIVOT_MAX_ACCELERATION_DEG_PER_SECOND_SQUARED));
+        // inverted
+        // encoder gear ratio
+        // current limit
+    }
+
+    public void initIntakeTalons() {
+        TalonFXConfigurator cfgFront = intakeTalonFront.getConfigurator();
+        TalonFXConfigurator cfgBack = intakeTalonFront.getConfigurator();
+
+        TalonFXConfiguration toApply = new TalonFXConfiguration();
+        toApply.CurrentLimits.SupplyCurrentLimit = IntakeConstants.INTAKING_TALONS_CURRENT_LIMIT_AMPS;
+        toApply.CurrentLimits.SupplyCurrentLimitEnable = true;
+        toApply.MotorOutput.NeutralMode = NeutralModeValue.Coast;
+
+        cfgFront.apply(toApply);
+        cfgBack.apply(toApply);
+
+        intakeTalonBack.setControl(intakeTalonBackFollower);
+    }
+
+    /** @return the last controlled time by the timer, if the lastControlledTime is empty (first record), then return 20 milliseconds
+     */
+    public double getLastControlledTime() {
+        if (lastControlledTime.isEmpty()) {
+            return Constants.PERIOD_TIME_MILLISECONDS;
+        }
+        return lastControlledTime.get();
+    }
+
+    /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+    public double getCosineArmAngle() {
+        return Math.cos(
+                Units.degreesToRadians(
+                        pivotEncoder.getPosition() - IntakeConstants.INTAKE_POSITION_WHEN_HORIZONTAL_DEGREES));
+    }
+
+    public double getOffsetAbsPositionDeg() {
+        return pivotAbsoluteEncoder.getPosition() + IntakeCal.INTAKE_ABSOLUTE_ENCODER_ZERO_OFFSET_DEG;
+      }
+
+    public void moveToPos(IntakePosition pos) {
+        pivotController.setGoal(intakePositionMap.get(pos));
+        double intakeDemandVoltsA = pivotController.calculate(getOffsetAbsPositionDeg());
```

Comment:
Here we control based on the absolute encoder, but then in `getCosineArmAngle` we use the relative encoder. We should use the same encoder for both

Reply:
changed to absolute for both